### ISSUE TITLE
[stable/datadog] unable to set valueFrom in extra env variables

### ISSUE THREAD

**Issue by zlangbert** [helm/charts#22267]
**Describe the bug**
Previously I was setting values like this to enable cluster check advanced dispatching:
```
clusterChecksRunner:
  enabled: true
  env:
    - name: DD_CLC_RUNNER_ENABLED
      value: "true"
    - name: DD_CLC_RUNNER_HOST
      valueFrom:
        fieldRef:
          fieldPath: status.podIP
```

But after https://github.com/helm/charts/pull/22006 landed to fix https://github.com/helm/charts/issues/21853 it will only render env variables that have a `value` and not `valueFrom`. This caused an env variable with a name and no value and the pods to crash loop.

**Version of Helm and Kubernetes**:
Helm 3.2.0, Kubernetes 1.16

**Which chart**:
stable/datadog

**What happened**:
After upgrading the chart to the latest version the value of the `DD_CLC_RUNNER_HOST` did not render and the pods went into a crash loop.

**What you expected to happen**:
That any type of environment variable could be set.

It would be great to restore the ability to use any environment variable value type needed.

/cc @clamoriniere

**Comment by josefschabasser** [comment]
Shameless bump!

**Comment by gajus** [comment]
Bump @clamoriniere @L3n41c @lucasreed

**Comment by josefschabasser** [comment]
Fixed by #23080
At least on my end :smile_cat:

**Comment by zlangbert** [comment]
Good catch @josefschabasser! Latest version of the chart fixes my issue as well, I think this can be closed.
